File: 1935.py
input = __import__('sys').stdin.readline

n = int(input())
op = input().rstrip()
num = [float(input()) for _ in range(n)]
ops = '+-*/'
stack = []
temp1, temp2 = 0.0, 0.0

for i in op:
    if i in ops:
        temp1 = stack.pop()
        temp2 = stack.pop()
        if i == '+':
            stack.append(temp2 + temp1)
        elif i == '-':
            stack.append(temp2 - temp1)
        elif i == '*':
            stack.append(temp2 * temp1)
        elif i == '/':
            stack.append(temp2 / temp1)
    else:
        stack.append(num[ord(i) - ord('A')])

print('%.2f' % stack[0])

File: 1935.cpp
#include <bits/stdc++.h>

using namespace std;
typedef long long ll;

int main() {
	int n;
	cin >> n;

	string s;
	cin >> s;

	vector<double> a;
	double temp;
	for (int i = 0; i < n; i++) {
		cin >> temp;
		a.push_back(temp);
	}

	stack<double> stc;
	double temp1, temp2;
	for (int i = 0; i < s.length(); i++) {
		switch (s[i]) {
			case '+':
				temp1 = stc.top();
				stc.pop();
				temp2 = stc.top();
				stc.pop();
				stc.push(temp2 + temp1);
				break;

			case '-':
				temp1 = stc.top();
				stc.pop();
				temp2 = stc.top();
				stc.pop();
				stc.push(temp2 - temp1);
				break;

			case '*':
				temp1 = stc.top();
				stc.pop();
				temp2 = stc.top();
				stc.pop();
				stc.push(temp2 * temp1);
				break;

			case '/':
				temp1 = stc.top();
				stc.pop();
				temp2 = stc.top();
				stc.pop();
				stc.push(temp2 / temp1);
				break;

			default:
				stc.push(a[s[i] - 'A']);
		}
	}

	printf("%.2f", stc.top());
}
